src/sinks/util/mod.rs

```diff
@@ -104,14 +106,315 @@ impl<I> EncodedEvent<I> {
+                    "KERN" => 0,
+                    "USER" => 1,
+                    "MAIL" => 2,
+                    "DAEMON" => 3,
+                    "AUTH" => 4,
+                    "SYSLOG" => 5,
+                    "LPR" => 6,
+                    "NEWS" => 7,
+                    "UUCP" => 8,
+                    "CRON" => 9,
+                    "AUTHPRIV" => 10,
+                    "FTP" => 11,
+                    "NTP" => 12,
+                    "SECURITY" => 13,
+                    "CONSOLE" => 14,
+                    "SOLARIS-CRON" => 15,
+                    "LOCAL0" => 16,
+                    "LOCAL1" => 17,
+                    "LOCAL2" => 18,
+                    "LOCAL3" => 19,
+                    "LOCAL4" => 20,
+                    "LOCAL5" => 21,
+                    "LOCAL6" => 22,
+                    "LOCAL7" => 23,
+                    _ => 24,
+                };
+                if num == 24 {
+                    return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"unknown facility"));
+                } else {
+                    return Ok(Facility::Fixed(num))
+                }
+            }
+        }
+    }
+}
+
+fn default_severity() -> Severity {
+    Severity::Fixed(0)
+}
+
+fn deserialize_severity<'de, D>(d: D) -> Result<Severity, D::Error>
+    where D: de::Deserializer<'de>
+{
+    let value: String = String::deserialize(d)?;
+    let num_value = value.parse::<u8>();
+    match num_value {
+        Ok(num) => {
+            if num > 6 {
+                return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"severity number too large"))
+            } else {
+                return Ok(Severity::Fixed(num))
+            }
+        }
+        Err(_) => {
+            if let Some(field_name) = value.strip_prefix("$.message.") {
+                return Ok(Severity::Field(field_name.to_string()));
+            } else {
+                let num = match value.to_uppercase().as_str() {
+                    "EMERGENCY" => 0,
+                    "ALERT" => 1,
+                    "CRITICAL" => 2,
+                    "ERROR" => 3,
+                    "WARNING" => 4,
+                    "INFORMATIONAL" => 5,
+                    "DEBUG" => 6,
+                    _ => 7,
+                };
+                if num > 6 {
+                    return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"unknown severity"))
+                } else {
+                    return Ok(Severity::Fixed(num))
+                }
+            }
+        }
+    }
+}
+
 /**
  * Enum representing different ways to encode events as they are sent into a Sink.
  */
-#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 #[serde(rename_all = "snake_case")]
 pub enum Encoding {
     Text,
     Json,
+    Syslog(SyslogConf),
+}
+
+fn add_log_source(log: &LogEvent, buf: &mut String) {
```

Comment:
This looks kubernetes specific.  That seems like a concern for a general syslog sink.  Do we need to abstract it someplace else or at least we should add  TODO to do something with it when we do an upstream pull

Reply:
This PR is not going upstream as is, including for this ^^^ reason.